src/util/psbt/map/output.rs

```diff
@@ -106,11 +121,11 @@ impl TapTree {
     ///
     /// # Return
     /// A `TapTree` iff the `inner` builder is complete, otherwise return the inner as `Err`.
-    pub fn from_inner(inner: TaprootBuilder) -> Result<Self, TaprootBuilder> {
+    pub fn from_inner(inner: TaprootBuilder) -> Result<Self, IncompleteTapTree> {
```

Comment:
It is rust convention to return `inner` as `Err` when it is consumed. So, the caller at least gets the copy it needs. 

I don't remember completely, but I think @Kixunil suggested I do this in his review.

Reply:
Ah, ok. But it is really confusing when working with an API like that - I was confused byself, as you've seen in the issue

Reply:
@sanket1729 what about making `IncompleteTapTree` type to be a wrapper around `TaprootBuilder`, implemeting `std::error::Error`?

Reply:
Reworked that way.

Reply:
I will let him comment on this.

Reply:
Invoking @Kixunil: please see above.

Reply:
FWIW, I like the implemented solution in this PR in case we decide to proceed with this.

Reply:
Yes, error wrapper containing original data is even more idiomatic Rust. [`PoisonError`](https://doc.rust-lang.org/std/sync/struct.PoisonError.html) is a good example of this.

It could be interesting to do it the way `std` does - instead of private field have accessor methods, so that the type can be extended. Or add a dummy field to emulate `#[non_exhaustive]` (and change it to `#[non_exhaustive]` after MSRV bump). Not sure how likely the change is here (writing this in hurry).